contracts/LimitsChecker.sol

```diff
@@ -0,0 +1,289 @@
+    constructor(
+        address[] memory _setLimitParameterRoleHolders,
+        address[] memory _updateSpentAmountRoleHolders,
+        IBokkyPooBahsDateTimeContract _bokkyPooBahsDateTimeContract
+    ) {
+        for (uint256 i = 0; i < _setLimitParameterRoleHolders.length; i++) {
+            _setupRole(SET_LIMIT_PARAMETERS_ROLE, _setLimitParameterRoleHolders[i]);
+        }
+        for (uint256 i = 0; i < _updateSpentAmountRoleHolders.length; i++) {
+            _setupRole(UPDATE_SPENT_AMOUNT_ROLE, _updateSpentAmountRoleHolders[i]);
+        }
+        bokkyPooBahsDateTimeContract = _bokkyPooBahsDateTimeContract;
+    }
+
+    // -------------
+    // EXTERNAL METHODS
+    // -------------
+
+    /// @notice Checks if _payoutAmount is less or equal than the may be spent
+    /// @param _payoutAmount Motion total amount
+    /// @param _motionDuration Motion duration - minimal time required to pass before enacting of motion
+    /// @return True if _payoutAmount is less or equal than may be spent
+    /// @dev note that upfront check is used to compare _paymentSum with total limit in case
+    /// when motion is started in one period and will be probably enacted in the next.
+    function isUnderSpendableBalance(uint256 _payoutAmount, uint256 _motionDuration)
+        external
+        view
+        returns (bool)
+    {
+        if (block.timestamp + _motionDuration >= currentPeriodEndTimestamp) {
+            return _payoutAmount <= limit;
+        } else {
+            return _payoutAmount <= _spendableBalance();
+        }
+    }
+
+    /// @notice Checks if _payoutAmount may be spent and increases spentAmount on _payoutAmount.
+    /// @notice Also updates the period boundaries if necessary.
+    function updateSpentAmount(uint256 _payoutAmount) external onlyRole(UPDATE_SPENT_AMOUNT_ROLE) {
+        /// When it is necessary to shift the currentPeriodEndTimestamp it takes on a new value.
+        /// And also spent is set to zero. Thus begins a new period.
+        if (block.timestamp >= currentPeriodEndTimestamp) {
+            currentPeriodEndTimestamp = uint128(_getPeriodEndFromTimestamp(block.timestamp));
+            spentAmount = 0;
+            emit CurrentPeriodAdvanced(currentPeriodEndTimestamp);
+        }
+
+        require(_payoutAmount <= _spendableBalance(), ERROR_SUM_EXCEEDS_SPENDABLE_BALANCE);
+        spentAmount += uint128(_payoutAmount);
+
+        (
+            uint256 _alreadySpentAmount,
+            uint256 _spendableBalanceInPeriod,
+            uint256 _periodStartTimestamp,
+            uint256 _periodEndTimestamp
+        ) = _getCurrentPeriodState();
+
+        emit SpendableAmountChanged(
+            _alreadySpentAmount,
+            _spendableBalanceInPeriod,
+            _periodStartTimestamp,
+            _periodEndTimestamp
+        );
+    }
+
+    /// @notice Returns balance that can be spent in the current period
+    /// @notice If period advanced and the period was not shifted,
```

Comment:
I would clarify what "the period was not shifted" means. Replace it to "no call to `updateSpentAmount` or `setLimitParameters` made", but dunno